src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -389,7 +389,7 @@ private bool NumberHandingIsApplicable()
                     else
                     {
                         // Slower path for non-generic types that implement IDictionary<string, object>.
-                        converter = JsonMetadataServices.ObjectConverter;
+                        converter = Options.GetConverterInternal(JsonTypeInfo.ObjectType);
```

Comment:
Any size savings if we do the same on L405 or is `JsonElement`/`JsonElementConverter` still rooted elsewhere?

Reply:
There is potential savings here by not rooting; pushed another commit that removes the reference to `JsonMetadataServices`.